core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/internal/CoreRemoteMongoClientImpl.java

```diff
@@ -62,21 +64,44 @@ public CoreRemoteMongoClientImpl(final CoreStitchServiceClient service,
     );
   }
 
-  @Override
-  public void onRebindEvent() {
-    if (!lastActiveUserId.equals(appInfo.getAuthMonitor().getActiveUserId())) {
-      this.lastActiveUserId = appInfo.getAuthMonitor().getActiveUserId() != null
-          ? appInfo.getAuthMonitor().getActiveUserId() : "";
+  private void onAuthEvent(final AuthEvent authEvent) {
+    switch (authEvent.getAuthEventType()) {
+      case USER_REMOVED:
+        final String userId = ((AuthEvent.UserRemoved)authEvent).getRemovedUser().getId();
+        if (!SyncMongoClientFactory.deleteDatabase(
+            appInfo,
+            service.getName(),
+            clientFactory,
+            userId
+        )) {
+          System.err.println("Could not delete database for user id " + userId);
+        }
+        break;
+      default:
+        if (!lastActiveUserId.equals(appInfo.getAuthMonitor().getActiveUserId())) {
+          this.lastActiveUserId = appInfo.getAuthMonitor().getActiveUserId() != null
+              ? appInfo.getAuthMonitor().getActiveUserId() : "";
 
-      // reinitialize the DataSynchronizer entirely.
-      // any auth event will trigger this.
-      this.dataSynchronizer.reinitialize(
-          SyncMongoClientFactory.getClient(
-              appInfo,
-              service.getName(),
-              clientFactory
-          )
-      );
+          // reinitialize the DataSynchronizer entirely.
+          // any auth event will trigger this.
+          this.dataSynchronizer.reinitialize(
```

Comment:
[q] if there is no active user after the active user has changed, shouldn't we call a different method that just stops the data synchronizer rather than reinitializes it? I see right now that the data synchronizer just ends up being bound to an "unbound" data directory. I guess that's fine but wouldn't it be cleaner if the data synchronizer just didn't run at all when logged out?

Reply:
It won't run anyway. This won't be an issue until we get to sync cadence. In the interim, this is fine.